lib/matplotlib/axes/_axes.py

```diff
@@ -390,6 +419,305 @@ def legend(self, *args, **kwargs):
+        rect = [xlim[0], ylim[0], xlim[1] - xlim[0], ylim[1] - ylim[0]]
+        rectpatch, connects = self.indicate_inset_bounds(
+                rect, inset_ax, **kwargs)
+
+        return rectpatch, connects
+
+    def secondary_xaxis(self, loc, conversion, **kwargs):
+        """
+        Add a second x-axis to this axes.
+
+        For example if we want to have a second scale for the data plotted on
+        the xaxis.
+
+        Parameters
+        ----------
+        loc : string or scalar
+            The position to put the secondary axis.  Strings can be 'top' or
+            'bottom', scalar can be a float indicating the relative position
+            on the axes to put the new axes (0 being the bottom, and 1.0 being
+            the top.)
+
+        conversion : tuple of floats or function
+            conversion between the parent xaxis values and the secondary xaxis
+            values.  If a tuple of floats, the floats are polynomial
+            co-efficients, with the first entry the highest exponent's
+            co-efficient (i.e. [2, 3, 1] is the same as
+            ``xnew = 2 x**2 + 3 * x + 1``).  If a function is specified it
+            should accept a float as input and return a float as the
+            result.
+
+        Other Parameters
+        ----------------
+        **kwargs : `~matplotlib.axes.Axes` properties.
+            Other miscellaneous axes parameters.
+
+        Returns
+        -------
+        ax : `~matplotlib.axes.Axes` (??)
+
+        """
+        secondary_ax = Secondary_Xaxis(self, loc, conversion, **kwargs)
+        self.add_child_axes(secondary_ax)
+        return secondary_ax
+
+
+    def secondary_yaxis(self, loc, conversion, **kwargs):
+        """
+        Add a second y-axis to this axes.
+
+        For example if we want to have a second scale for the data plotted on
+        the xaxis.
+
+        Parameters
+        ----------
+        loc : string or scalar
+            FIX
+            The position to put the secondary axis.  Strings can be 'top' or
+            'bottom', scalar can be a float indicating the relative position
+            on the axes to put the new axes (0 being the bottom, and 1.0 being
+            the top.)
+
+        conversion : tuple of floats or function
+            conversion between the parent xaxis values and the secondary xaxis
+            values.  If a tuple of floats, the floats are polynomial
+            co-efficients, with the first entry the highest exponent's
+            co-efficient (i.e. [2, 3, 1] is the same as
+            ``xnew = 2 x**2 + 3 * x + 1``).  If a function is specified it
+            should accept a float as input and return a float as the
+            result.
+
+        Other Parameters
+        ----------------
+        **kwargs : `~matplotlib.axes.Axes` properties.
+            Other miscellaneous axes parameters.
+
+        Returns
+        -------
+        ax : `~matplotlib.axes.Axes` (??)
```

Comment:
Why is that axes private? If you include it as a public object you could directly refer to it as the return type.

Reply:
It wasn't meant to be private.  I think the (??) was just because I wasn't sure that the ref was right... (sorry, this isn't a fully polished PR yet, just wanted to get feedback on the overall API)...

Reply:
I'm just wondering if having this as private object only would make it harder to customize, if needed.

Reply:
I'm not following what makes it private?  Its not meant to be private...

Reply:
What is returned is a `matplotlib.axes._secondary_axes.Secondary_Yaxis`. So one would expect to see something like 
```
Returns
-------
ax : `~matplotlib.axes._secondary_axes.Secondary_Yaxis`
```
right?

But that may not be possible due to `_secondary_axes` being "private" and not part of the documentation.

Reply:
Ah I see what you mean.  I find this sort of thing very confusing.  Somehow Axes is defined in a private sub module, and I think that’s what should be done here too. But I’m not clear on how the API bubbles up to be public.